src/dev/build/tasks/create_cdn_assets_task.ts

```diff
@@ -38,17 +38,27 @@ export const CreateCdnAssets: Task = {
       const manifest = Jsonc.parse(readFileSync(path, 'utf8')) as any;
       if (manifest?.plugin?.id) {
         const pluginRoot = resolve(dirname(path));
+        const assetsSources = [
+          resolve(pluginRoot, 'public', 'assets'),
+          resolve(pluginRoot, 'assets'),
+        ];
+        // packages/core/apps/core-apps-server-internal/src/core_app.ts
+        const assetsDest = resolve(assets, buildSha, 'plugins', manifest.plugin.id, 'assets');
 
-        try {
-          // packages/core/plugins/core-plugins-server-internal/src/plugins_service.ts
-          const assetsSource = resolve(pluginRoot, 'assets');
-          const assetsDest = resolve('plugins', manifest.plugin.id, 'assets');
-          await access(assetsSource);
-          await mkdirp(assetsDest);
-          await copyAll(assetsSource, assetsDest);
-        } catch (e) {
-          // assets are optional
-          if (!(e.code === 'ENOENT' && e.syscall === 'access')) throw e;
+        for (const assetsSource of assetsSources) {
+          try {
+            await access(assetsSource);
+            await mkdirp(assetsDest);
+            await copyAll(assetsSource, assetsDest);
+          } catch (e) {
+            if (e.code === 'ENOENT' && e.syscall === 'access') {
+              // miss, check the next asset source path
+              continue;
+            }
+            throw e;
+          }
+          // found and copied assets, no need to look further
+          break;
```

Comment:
This is a little strange end user DX, but I figured both `assets` and `public/assets` are allowed. We could remove this line and just copy from both?

Reply:
Wouldn't it look nicer, if you put the break; right after the successful copy?

Reply:
OK, based on [this code](https://github.com/elastic/kibana/blob/b3ab7584329fd8a0ac64393c541114ee1eda56c9/packages/core/plugins/core-plugins-server-internal/src/plugins_service.ts#L320) I think I'm wrong about how asset discovery works. Looks like we only read from `public/assets`. I'll double check with someone from Core :)